proposals/0000-slurp.rst

```diff
@@ -0,0 +1,454 @@
+
+-  Everyone wants Haskell to succeed and be widely used
+-  Everyone wants Haskell users to have a frictionless experience of
+   using libraries written by others
+-  Everyone is trying hard to do the Right Thing
+
+That makes it tantalising that we are having difficulty finding a
+solution. But it also makes us optimistic that we can find one if we
+work together.
+
+Appendix A: SLURP governance and terms of use
+---------------------------------------------
+
+The below is an example policy document. It should be discussed
+independently of the technical proposal above: none of the below imposes
+any additional constraints on the implementation.
+
+Governance
+~~~~~~~~~~
+
+SLURP will be governed by the *SLURP trustees*.
+
+The SLURP trustees include representatives from the various stakeholders
+whose composition will change as the ecosystems around SLURP changes.
+Initially, we propose the SLURP trustees to be comprised of (1) one
+representative of the Hackage maintainers, (2) one representative of the
+Stackage curators, and (3) one member-at-large from the general Haskell
+community. The number of trustees can change over time. The
+member-at-large will chair the group of trustees.
+
+
+Changes to the composition of the SLURP trustees are decided by the SLURP trustees and need to be unanimous. All other decisions are made by simple majority vote if there is a dispute among the trustees.
+This governance mechanism is, by design, lightweight; for example, the number of trustees is small.  If, in the light of experience, it seems that a more substantial mechanism is needed, we can revisit the issue as a community.
+
+Terms of use
+~~~~~~~~~~~~
+
+Users of SLURP agree to abide by the rules stated in the following. IP
+addresses or IP ranges of users who repeatedly violate the terms of use
+may be blocked. This decision is made by the SLURP trustees.
+
+§1 Valid SLURP entries
+^^^^^^^^^^^^^^^^^^^^^^
+
+Any entry added with ``PUT /package/:pkgname`` that maps the new package
+name *pkgname* to the URL *url* must abide by the following
+restrictions:
+
+-  Both the *pkgname* and the *url* must not already exist in the
+   registry.
+-  The *pkgname* must not be unlawful, threatening, abusive, libelous,
+   defamatory, obscene, offensive, indecent, pornographic, profane, or
+   otherwise objectionable.
+-  The endpoint identified by *url* must conform to the specification
+   given in the section entitled ”The package API”, and must continue to
+   do so over time. Endpoints that no longer conform to the
+   specification are liable to removal from the SLURP registry after one
+   month.
+-  The tarballs that can be obtained via that package API need to be
+   genuine Haskell packages. This means that the tarballs must include
+   package metadata and a build file that conforms to Cabal-the-spec.
+   This metadata must (1) identify the packages as *pkgname* and (2)
+   include valid license and copyright information.
```

Comment:
Can we require the URL to be a `https:` url? (I.e. not `http`, not `gopher`, not `ftp`, not `git`)